docs/book/ssh-and-sudo-authorization.md

```diff
@@ -256,85 +263,137 @@ able to login.
+# of an internal elevation system.
 allow {
-  input.host_identity.host_id = "frontend"
-  input.user = "frontend-dev"
+  elevate.tickets[sysinfo.pam_username] = display_responses.ticket
 }
 EOF
 ```
 
 Load this policy into OPA.
 
 ```shell
-curl -X PUT --data-binary @sudo_authz_special.rego \
-  localhost:8181/v1/policies/sudo_authz_special
+curl -X PUT --data-binary @sudo_authz_elevated.rego \
+  localhost:8181/v1/policies/sudo_authz_elevated
 ```
 
-Confirm that the user `frontend-dev` can login to the `frontend` host and run sudo commands.
+Confirm that the user `frontend-dev` can indeed use `sudo`.
 
 ```shell
 ssh -p 2222 frontend-dev@localhost \
-  -i ssh_authz.key -o StrictHostKeyChecking=no
+  -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
+
 sudo ls /
-exit
 ```
+You should be prompted with the message that we defined in our *display* policy
+for both the SSH and `sudo` authorization cycles.
+This happens because the *display* policy is shared by the PAM configurations of SSH and `sudo`.
+In production, it is more practical to use separate policy packages for each PAM configuration.
 
-Lastly, remove the policy from OPA and confirm the user's original rights are restored.
 
-```shell
-curl -i -X DELETE localhost:8181/v1/policies/sudo_authz_special
-```
+For SSH, the value entered does not matter.
+For `sudo`, enter the ticket number `1234` to get access.
+
+Lastly, update the mocked elevation API and confirm the user's original rights are restored.
 
 ```shell
-ssh -p 2222 frontend-dev@localhost \
-  -i ssh_authz.key -o StrictHostKeyChecking=no
-sudo ls /
-exit
+curl -X PUT localhost:8181/v1/data/elevate -d \
+'{
+  "tickets": {}
+}'
 ```
 
+You will find that running `sudo ls /` as the `frontend-dev` user is disallowed again.
+
+It is possible to configure the *display* policy to only make the PAM module prompt for the
+elevation ticket when our mock API has a non-empty `tickets` object. So when there are no
+elevated users, there will be no prompt for a ticket. Doing this has been left as an
+exercise for the reader.
+
 ## Wrap Up
 
 Congratulations for finishing the tutorial!
 
  You learned a number of things about SSH with OPA:
 
-* OPA gives you fine-grained policy control over SSH and sudo on any host where you've
-  installed OPA and its PAM module.
-* You write allow/deny policies to control who has access to what.
-* You can import external data into OPA and write policies that depend on.
-  that data.
+* OPA gives you fine-grained policy control over SSH, `sudo`, and any other application that uses PAM,
+  on any host where you've
+  installed OPA and its PAM module. Although this tutorial used the some of the same policies for both
+  SSH and sudo, you should use separate, fine-grained policies for each application that supports PAM.
+* You write allow/deny policies to control who has access to what, based on authorization context generated
```

Comment:
Would rewrite: "You write allow/deny policies to control who has access to what, based on authorization context generated using steps defined in other policies." to 

"Writing allow/deny policies to control who has access to what using context from the user, servers, and a ticketing system."

Reply:
Done, except for the ticketing system bit which is generalized in the next bullet point.

Reply:
Awesome, even better